sympy/core/symbol.py

```diff
@@ -697,16 +700,16 @@ def symbols(names, *, cls=Symbol, **args):
 
     if isinstance(names, str):
         marker = 0
-        literals = [r'\,', r'\:', r'\ ']
-        for i in range(len(literals)):
-            lit = literals.pop(0)
-            if lit in names:
+        splitters = r'\,', r'\:', r'\ '
+        literals: List[tTuple[str, str]] = []
```

Comment:
I will never be a big supporter of this. While I may see the point of functions having (well-)defined inputs and outputs, bloating the internals like this... (An end-of-line comment would be better from a readability point-of-view, if it is now a problem to understand how it works. Though I understand that there are possible advantages for static code checkers to use this.)

So let us just say that I do not like it, but will not block this.

Reply:
I like the remaining though!